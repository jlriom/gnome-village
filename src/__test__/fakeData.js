export const category = {
    category: "category",
    count: 0
}

export const habitantsByHairColor = [
    {
        hair_color: 'Black',
        count: 256
    },
    {
        hair_color: 'Gray',
        count: 258
    },
    {
        hair_color: 'Green',
        count: 267
    },
    {
        hair_color: 'Pink',
        count: 285
    },
    {
        hair_color: 'Red',
        count: 273
    }
];

export const habitantsByProfession = [
    {
        profession: 'Baker',
        count: 214
    },
    {
        profession: 'Blacksmith',
        count: 180
    },
    {
        profession: 'Brewer',
        count: 200
    },
    {
        profession: 'Butcher',
        count: 210
    },
    {
        profession: 'Carpenter',
        count: 216
    },
    {
        profession: 'Cook',
        count: 190
    },
    {
        profession: 'Farmer',
        count: 205
    },
    {
        profession: 'Gemcutter',
        count: 186
    },
    {
        profession: 'Leatherworker',
        count: 185
    },
    {
        profession: 'Marble Carver',
        count: 190
    },
    {
        profession: 'Mason',
        count: 214
    },
    {
        profession: 'Mechanic',
        count: 182
    },
    {
        profession: 'Medic',
        count: 187
    },
    {
        profession: 'Metalworker',
        count: 192
    },
    {
        profession: 'Miner',
        count: 201
    },
    {
        profession: 'Potter',
        count: 202
    },
    {
        profession: 'Prospector',
        count: 202
    },
    {
        profession: 'Sculptor',
        count: 182
    },
    {
        profession: 'Smelter',
        count: 203
    },
    {
        profession: 'Stonecarver',
        count: 187
    },
    {
        profession: 'Tailor',
        count: 199
    },
    {
        profession: 'Tax inspector',
        count: 192
    },
    {
        profession: ' Tinker',
        count: 202
    },
    {
        profession: 'Woodcarver',
        count: 199
    }
];

export const auth = { name: "name", isGuest: false };

export const habitant = {
    id: 362,
    name: 'Cogwitz Bittergrill',
    thumbnail: 'http://www.publicdomainpictures.net/pictures/120000/nahled/white-hen.jpg',
    age: 127,
    weight: 40.074535,
    height: 93.66314,
    hair_color: 'Black',
    professions: [
        'Prospector',
        ' Tinker',
        'Baker',
        'Tailor',
        'Leatherworker',
        'Sculptor'
    ],
    friends: [
        'Whitwright Fussbuster',
        'Fizwood Bitterrocket',
        'Fizwood Felpower'
    ]
};

export const friends = [
    {
        id: 363,
        name: 'Cogwitzs friend ',
        thumbnail: 'http://www.publicdomainpictures.net/pictures/120000/nahled/white-hen.jpg',
        age: 127,
        weight: 40.074535,
        height: 93.66314,
        hair_color: 'Black',
        professions: [
            'Prospector',
            ' Tinker'
        ],
        friends: [
            'Whitwright Fussbuster'
        ]
    }
];
