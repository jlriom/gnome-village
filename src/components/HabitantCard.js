import React from "react";
import Card from "react-bootstrap/Card";
import LazyLoad from "react-lazyload";

const ProfessionItem = ({ profession }) => (
	<>
		<dt className='col-sm-6'>&nbsp;</dt>
		<dd className='col-sm-6 '>{profession}</dd>
	</>
);

const HabitantCard = ({ habitant }) => {
	const {
		name,
		thumbnail,
		age,
		weight,
		height,
		hair_color,
		professions
	} = habitant;

	return (
		<>
			<Card style={{ width: "18rem" }}>
				<LazyLoad once>
					<Card.Img variant='top' src={thumbnail} alt={name} />
				</LazyLoad>
				<Card.Body>
					<Card.Title>{name}</Card.Title>
					<div className='container' style={{ weight: "100%" }}>
						<dl className='row'>
							<dt className='col-sm-6'>Age:</dt>
							<dd className='col-sm-6'> {age}</dd>
							<dt className='col-sm-6'>Hair Color:</dt>
							<dd className='col-sm-6'> {hair_color}</dd>
							<dt className='col-sm-6'>Weight:</dt>
							<dd className='col-sm-6'> {weight.toFixed(3)}</dd>
							<dt className='col-sm-6'>Height:</dt>
							<dd className='col-sm-6'> {height.toFixed(3)}</dd>
						</dl>
						<hr></hr>
						<div className='row'>
							{professions && professions.length > 0 ? (
								<dl className='row'>
									<dt className='col-sm-6'>Professions:</dt>
									<dd className='col-sm-6'> &nbsp;</dd>
									{professions.map((profession, index) => (
										<ProfessionItem key={index} profession={profession} />
									))}
								</dl>
							) : (
								<span></span>
							)}
						</div>
					</div>
				</Card.Body>
			</Card>
		</>
	);
};

export default HabitantCard;
