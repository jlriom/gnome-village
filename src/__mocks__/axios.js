import * as fakeData from "./fakeData";

export default {
	get: jest.fn(() =>
		Promise.resolve({
			data: fakeData.habitants
		})
	),
	post: jest.fn(() => Promise.resolve({ data: fakeData.habitant })),
	put: jest.fn(() => Promise.resolve({ data: fakeData.habitant })),
	delete: jest.fn(() => Promise.resolve({ data: fakeData.habitant }))
};
