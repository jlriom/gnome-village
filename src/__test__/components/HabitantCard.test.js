import React from "react";
import HabitantCard from "../../components/HabitantCard";
import renderer from "react-test-renderer";
import { habitant } from "../../__mocks__/fakeData";

describe("<HabitantCard />", () => {
	it("renders ok", () => {
		const tree = renderer.create(<HabitantCard habitant={habitant} />);
		expect(tree).toMatchSnapshot();
	});
});
