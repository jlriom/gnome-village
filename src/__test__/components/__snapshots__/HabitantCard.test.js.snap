// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HabitantCard /> renders ok 1`] = `
<div
  className="card"
  style={
    Object {
      "width": "18rem",
    }
  }
>
  <div
    className="lazyload-placeholder"
    style={
      Object {
        "height": undefined,
      }
    }
  />
  <div
    className="card-body"
  >
    <div
      className="card-title h5"
    >
      Cogwitz Bittergrill
    </div>
    <div
      className="container"
      style={
        Object {
          "weight": "100%",
        }
      }
    >
      <dl
        className="row"
      >
        <dt
          className="col-sm-6"
        >
           
          Age:
        </dt>
        <dd
          className="col-sm-6"
        >
           
          127
        </dd>
        <dt
          className="col-sm-6"
        >
           
          Hair Color:
        </dt>
        <dd
          className="col-sm-6"
        >
           
          Black
        </dd>
        <dt
          className="col-sm-6"
        >
           
          Weight:
        </dt>
        <dd
          className="col-sm-6"
        >
           
          40.075
        </dd>
        <dt
          className="col-sm-6"
        >
           
          Height:
        </dt>
        <dd
          className="col-sm-6"
        >
           
          93.663
        </dd>
      </dl>
      <hr />
      <div
        className="row"
      >
        <dl
          className="row"
        >
          <dt
            className="col-sm-6"
          >
            Professions:
          </dt>
          <dd
            className="col-sm-6"
          >
              
          </dd>
          <dt
            className="col-sm-6"
          >
             
          </dt>
          <dd
            className="col-sm-6 "
          >
            Prospector
          </dd>
          <dt
            className="col-sm-6"
          >
             
          </dt>
          <dd
            className="col-sm-6 "
          >
             Tinker
          </dd>
          <dt
            className="col-sm-6"
          >
             
          </dt>
          <dd
            className="col-sm-6 "
          >
            Baker
          </dd>
          <dt
            className="col-sm-6"
          >
             
          </dt>
          <dd
            className="col-sm-6 "
          >
            Tailor
          </dd>
          <dt
            className="col-sm-6"
          >
             
          </dt>
          <dd
            className="col-sm-6 "
          >
            Leatherworker
          </dd>
          <dt
            className="col-sm-6"
          >
             
          </dt>
          <dd
            className="col-sm-6 "
          >
            Sculptor
          </dd>
        </dl>
      </div>
    </div>
  </div>
</div>
`;
